src/hmem_cuda_gdrcopy.c

```diff
@@ -292,6 +293,44 @@ void cuda_gdrcopy_from_dev(uint64_t handle, void *hostptr,
 			  GDRCOPY_FROM_DEVICE);
 }
 
+static ssize_t cuda_gdrcopy_iov_buf(uint64_t handle,
+                                    const struct iovec *iov, size_t iov_count,
+                                    uint64_t iov_offset, void *buf,
+                                    size_t size, enum gdrcopy_dir dir)
+{
+	uint64_t done = 0, len;
+	char *dev_buf;
+	size_t i;
+
+	for (i = 0; i < iov_count && size; i++) {
+		len = ofi_iov_bytes_to_copy(&iov[i], &size, &iov_offset, &dev_buf);
+
+		if (!len)
+			continue;
+
+		cuda_gdrcopy_impl(handle, dev_buf, (void *)((char *)buf + done), len, dir);
+
+		done += len;
+	}
+	return done;
+}
+
+ssize_t cuda_gdrcopy_to_dev_iov(uint64_t handle, const struct iovec *iov,
```

Comment:
if we are exposing these functions to providers, I would like the adjust name to `ofi_gdrcopy_to/from_cuda_iov`

Reply:
Good idea. Addressed.